### PR TITLE
refactor: QueryChangeStreamAction

### PR DESCRIPTION
**Pull request by zoercai**
Extracted QueryChangeStreamAction and reordered the action fields.

### CODE REVIEW

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFn.java [hunk 1]
```diff
@@ -120,53 +117,25 @@ public void setup() {
...
     this.deletePartitionAction = actionFactory.deletePartitionAction(partitionMetadataDao);
     this.donePartitionAction = actionFactory.donePartitionAction();
-
-    this.dataChangeRecordAction = actionFactory.dataChangeRecordAction();
-    this.heartbeatRecordAction = actionFactory.heartbeatRecordAction();
-    this.childPartitionsRecordAction =
-        actionFactory.childPartitionsRecordAction(partitionMetadataDao);
-  }
-
-  // TODO: Close DAOs on teardown
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFn.java:137]: Could you leave this `TODO` in this class?

File: sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFnTest.java [hunk 2]
```diff
@@ -106,15 +108,22 @@ public void setUp() {
     final ActionFactory actionFactory = mock(ActionFactory.class);
     final PartitionMetadataDao partitionMetadataDao = mock(PartitionMetadataDao.class);
     changeStreamDao = mock(ChangeStreamDao.class);
+    changeStreamRecordMapper = mock(ChangeStreamRecordMapper.class);
+    dataChangeRecordAction = mock(DataChangeRecordAction.class);
+    heartbeatRecordAction = mock(HeartbeatRecordAction.class);
+    childPartitionsRecordAction = mock(ChildPartitionsRecordAction.class);
+    queryChangeStreamAction =
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFnTest.java:115]: Could you move the tests that were exercising the refactored class to its own test class? (And add new tests here that use `queryChangeStreamAction = mock(QueryChangeStreamAction.class)`?)

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFn.java (hunk 1 above)
Comment (zoercai) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFn.java:137]: Oh I haven't removed it, it's still there on line 179 :)

Comment (thiagotnunes) [reply sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFn.java:137]: Ah thanks!

File: sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFn.java [hunk 3]
```diff
@@ -68,13 +63,15 @@
   private transient ChangeStreamRecordMapper changeStreamRecordMapper;
   private transient ChangeStreamDao changeStreamDao;
 
+  private transient DataChangeRecordAction dataChangeRecordAction;
```
Comment (thiagotnunes) [inline sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/cdc/ReadChangeStreamPartitionDoFn.java:66]: Do these need to be fields? Or can they be local variables on the `setup` method?

**Review by thiagotnunes — APPROVED** [review]

**Pull request merged**